### PR TITLE
add basket tag

### PR DESCRIPTION
**Pull request by diarmuidmcg**

### CODE REVIEW

**Comment by diarmuidmcg** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/miamtech/MealzNavModuleIOS/46?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#46** <a href="https://app.graphite.dev/github/pr/miamtech/MealzNavModuleIOS/46?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#45** <a href="https://app.graphite.dev/github/pr/miamtech/MealzNavModuleIOS/45?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#44** <a href="https://app.graphite.dev/github/pr/miamtech/MealzNavModuleIOS/44?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#43** <a href="https://app.graphite.dev/github/pr/miamtech/MealzNavModuleIOS/43?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @diarmuidmcg and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/BasketTagFeatureConstructor.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//
+//  BasketTagFeatureConstructor.swift
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import Foundation
+import MealzUIModuleIOS
+import MiamIOSFramework
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/BasketTagFeatureConstructor.swift:10]:
```suggestion
import MealzIOSFramework
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/BasketTagFeatureNavCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+//  BasketTagFeatureNavCoordinator.swift
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import Foundation
+import MealzUIModuleIOS
+import miamCore
+import MiamIOSFramework
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/BasketTagFeatureNavCoordinator.swift:11]:
```suggestion
import mealzcore
import MealzIOSFramework
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/BasketTagFeatureNavCoordinatorProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//  BasketTagFeatureNavCoordinatorProtocol.swift
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import SwiftUI
+import MealzUIModuleIOS
+import MiamIOSFramework
+import miamCore
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/BasketTagFeatureNavCoordinatorProtocol.swift:11]:
```suggestion
import MealzIOSFramework
import mealzcore
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//  MealzBasketTagFeatureSwiftUI.swift
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import UIKit
+import SwiftUI
+import MealzUIModuleIOS
+import miamCore
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift:11]:
```suggestion
import MealzUIModuleIOS
import mealzcore
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureUIKit.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
+//
+//  MealzBasketTagFeatureUIKit.swift
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import UIKit
+import miamCore
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureUIKit.swift:9]:
```suggestion
import mealzcore
```

File: Sources/MealzNavModuleIOS/ViewController/GeneralViews/BasketTagViewController.swift [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import UIKit
+import SwiftUI
+import MiamIOSFramework
+import MealzUIModuleIOS
+import miamCore
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/ViewController/GeneralViews/BasketTagViewController.swift:12]:
```suggestion
import MealzIOSFramework
import MealzUIModuleIOS
import mealzore
```

File: Sources/MealzNavModuleIOS/ViewController/GeneralViews/BasketTagViewController.swift [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+//
+//
+//  Created by Diarmuid McGonagle on 12/03/2024.
+//
+
+import UIKit
+import SwiftUI
+import MealzIOSFramework
+import MealzUIModuleIOS
+import mealzore
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/ViewController/GeneralViews/BasketTagViewController.swift:12]:
```suggestion
import mealzcore
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                navigationController: navController
+            ),
+            recipeDetailsConstructor: recipeDetailsConstructor,
+            basketTagFeatureConstructor: basketTagFeatureConstructor
+        )
+    }
+    
+    public func makeUIViewController(context: Context) -> UIViewController {
+        coordinator.setBasketTag(retailerProductId: retailerProductId, scrollAlignment: scrollAlignment)
+        if let recipeCarpuselViewController = coordinator.navigationController.viewControllers.first {
```
Comment (walerowicz-miam) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift:41]: recipeCarouselViewController

**Review by walerowicz-miam — CHANGES_REQUESTED** [review]

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift (hunk 8 above)
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift:41]:
```suggestion
        if let recipeCarouselViewController = coordinator.navigationController.viewControllers.first {
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            return recipeCarpuselViewController
```
Comment (diarmuidmcg) [inline Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift:42]:
```suggestion
            return recipeCarouselViewController
```

File: Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift (hunk 8 above)
Comment (diarmuidmcg) [reply Sources/MealzNavModuleIOS/NavCoordinators/BasketTag/MealzBasketTagFeatureSwiftUI.swift]: c'est fait

**Pull request merged**